<!DOCTYPE html>
<!--
    Plain-Academic by Vasilios Mavroudis
    Released under the Simplified BSD License/FreeBSD (2-clause) License.
    https://github.com/mavroudisv/plain-academic
-->

<html lang="en">
<head>
  <title>Thomas Hamelryck</title>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css">
  <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.0/jquery.min.js"></script>
  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/js/bootstrap.min.js"></script>
  <link href='https://fonts.googleapis.com/css?family=Oswald:700' rel='stylesheet' type='text/css'>
</head>
<body>

<style>
    .aligncenter { text-align: center; }
</style>


<!-- Navigation -->
	<nav class="navbar navbar-inverse navbar-static-top" role="navigation">
	  <div class="container">
		<div class="navbar-header">
		  <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
						<span class="sr-only">Toggle navigation</span>
						<span class="icon-bar"></span>
						<span class="icon-bar"></span>
						<span class="icon-bar"></span>
					</button>
		</div>

		<!-- Collect the nav links, forms, and other content for toggling -->
		<div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
		  <ul class="nav navbar-nav">
				  <li><a href="index.html">Home</a></li>
				  <li><a href="research.html">Research</a></li>
				  <li><a href="group.html">Group</a></li> 
				  <li><a href="funding.html">Funding</a></li>
				  <li><a href="software.html">Software</a></li>
				  <li><a href="publications.html">Publications</a></li> 
				  <li><a href="thoughts.html">Thoughts</a></li> 
		  </ul>
		</div>
	  </div>
	</nav>
  
  <!-- Page Content -->
    <div class="container">

        <div class="row">

            <!-- Entries Column -->
            <div class="col-md-8">
                
                <!-- Main Image -->
                <img class="img-responsive" src="torus_900_300.png" alt=""><br>

                <div style="margin-top:3%; text-align:justify;">                

                Below is an outline of my research over the years and some key publications. <br> 
                For a full list of references, see <a href="https://scholar.google.dk/citations?user=YoTlzjkAAAAJ&hl=en&oi=ao">Google Scholar</a>.

                <p>
                <p>
                <b>Picture</b>: points sampled from the bivariate von Mises distribution on the torus.
                This distribution belongs to directional statistics - a branch of statistics that deals 
                with non-Euclidean data, such as for example dihedral angles in biomolecules.  
                

                <h3> Structural biology </h3>

                <p> I did my PhD at the <a href="http://www.vub.ac.be">Free University of Brussels (VUB), Belgium</a> on 
                the structural biology (using <b>X-ray crystallography</b>) of <b>proteins that 
                bind complex sugars (lectins)</b>, using
                the legume lectin family as a model system. Here are some of the key papers from that period. They
                shed light on how lectins distinguish between highly similar complex sugars, on the interplay between
                sugar binding and the quaternary structure of proteins, and on the structure of a plant-defense protein
                related to the legume lectins, called arcelin-5. The review that appeared in BBA in 1998 is still a 
                widely-read paper on the subject.  
                <ul>
                    <li>Hamelryck, T., Dao-Thi, MH., Poortmans, F., Chrispeels, MJ., Wyns, L., Loris, R. (1996). 
                    <b>The crystallographic structure of phytohemagglutinin-L</b>. 
                    <a href="https://www.jbc.org/article/S0021-9258(19)73856-3/fulltext">
                    Journal of Biological Chemistry, 271(34), 20479–20485.
                    </a>

                    <li>Hamelryck, T., Poortmans, F., Goossens, A., Angenon, G., Van Montagu, M., Wyns, L., Loris, R. (1996). 
                    <b>Crystal structure of Arcelin-5, a lectin-like defence protein from the seeds of Phaseolus vulgaris.</b>
                     J. Biol. Chem., 271, 32796-32802.

                    <li> 
                    Loris, R., Hamelryck, T., Bouckaert, J., & Wyns, L. (1998). 
                    <b>Legume lectin structure</b>. 
                    <a href="https://www.sciencedirect.com/science/article/abs/pii/S0167483897001829"> 
                    Biochimica et Biophysica Acta (BBA) - Protein Structure and Molecular Enzymology, 1383(1), 9–36. 
                    </a>

                    <li> 
                    Hamelryck, T., Loris, R., Bouckaert, J., Dao-Thi, MH., Strecker, G., Imberty, A., Fernandez, E., Wyns, L., Etzler, ME. (1999). 
                    <b>Carbohydrate binding, quaternary structure and a novel hydrophobic binding site in two legume lectin oligomers from Dolichos biflorus</b>. 
                    <a href="https://people.binf.ku.dk/~thamelry/publications/jmb_dbl.pdf">
                    Journal of Molecular Biology, 286(4), 1161–1177.
                    </a> 
                    
                    <li>
                    Hamelryck, T. W., Moore, J. G., Chrispeels, M. J., Loris, R., & Wyns, L. (2000). 
                    <b>The role of weak protein-protein interactions in multivalent lectin-carbohydrate binding: crystal structure of cross-linked FRIL</b>. 
                    <a href="https://people.binf.ku.dk/~thamelry/publications/jmb_fril.pdf">
                    Journal of Molecular Biology, 299(4), 875–883
                    </a>
                </ul>

                <p class="aligncenter">
                    <img src="dbl_lectin.png" alt="Dolichos biflorus lectin">
                </p>

                <h3> Structural bioinformatics </h3>
                After my PhD in protein X-ray crystallography, I turned to structural bioinformatics. 
                The Proteins (2005) article was my first publication in the field: a new way to look at <b>solvent exposure</b>, based on the surprising                
                observation that the environment around the side chain of an amino acid differs radically from its opposite side with respect to the distribition of the                number of neighbors. The result, <b>Half Sphere Exposure (HSE)</b>, is now an established method to quantify solvent exposure of an amino acid, next to 
                contact number and accessible surface area. I also developed <b><a href="https://biopython.org/wiki/The_Biopython_Structural_Bioinformatics_FAQ">Biopython's Bio.PDB library</a></b> 
                - still one of the standard tools to extract data from PDB files. 

                <p>
                <ul>
                    <li> Hamelryck, T. (2005). 
                    <b>An amino acid has two sides: a new 2D measure provides a different view of solvent exposure.</b>
                    Proteins: Structure, Function, and Bioinformatics, 59(1), 38–48.

                    <li> Hamelryck, T., Manderick, B. (2003). 
                    <b>PDB file parser and structure class implemented in Python</b>. 
                    Bioinformatics, 19(17), 2308–2310.
                    
                    <li> Cock, P. J., Antao, T., Chang, J. T., Chapman, B. A., Cox, C. J., Dalke, A., Friedberg, I., Hamelryck, T., Kauff, F., 
                    Wilczynski, B., and others. (2009). 
                    <b>Biopython: freely available Python tools for computational molecular biology and bioinformatics.</b>
                    Bioinformatics, 25(11), 1422–1423. 
                </ul>

                <p class="aligncenter">
                    <img src="biopdb_example.png" alt="Dolichos biflorus lectin" style="width: 90vw; min-width: 100px; max-width: 700px">
                </p>

                <p>

                <h3> Statistical structural bioinformatics </h3>
                Combining <b>Bayesian networks and directional statistics</b>, I worked on formulating <b>proper probabilistic models 
                of protein and RNA structure</b>. This work is still relevant for moving beyond point predictions obtained from 
                non-probabilistic deep network heuristics, I think. Much of this work has been done with my two close collaborators 
                at the University of Leeds, <a href="https://www1.maths.leeds.ac.uk/~sta6kvm/">Prof. Kanti KV. Mardia</a> 
                and <a href="https://eps.leeds.ac.uk/maths/staff/4044/professor-john-t-kent">Prof. John T. Kent</a>, 
                and my former PhD student (now professor) <a href="https://di.ku.dk/Ansatte/?pure=da/persons/275482">Wouter Boomsma</a>. 
                The evolutionary model published in 2017 was developed together with 
                <a href="https://www.stats.ox.ac.uk/people/jotun-hein">Prof. Jotun Hein</a>, University of Oxford and his PhD student Michael Golden.
                
                <p>
                <ul>
                    <li>Hamelryck, T., Kent, JT., Krogh, A. (2006). 
                    <b>Sampling realistic protein conformations using local structural bias</b>.
                    PLoS Computational Biology, 2: e131.

                    <li> Boomsma, W., Mardia, KV., Taylor, CC., Ferkinghoff-Borg, J., Krogh, A., Hamelryck, T. (2008). 
                    <b>A generative, probabilistic model of local protein structure</b>.
                    Proceedings of the National Academy of Sciences, 105(26), 8932–8937.

                    <li> Frellsen, J., Moltke, I., Thiim, M., Mardia, K. V., Ferkinghoff-Borg, J., Hamelryck, T. (2009). 
                    <b>A probabilistic model of RNA conformational space</b>.
                    PLoS Computational Biology, 5(6), e1000406.

                    <li> Harder, T., Boomsma, W., Paluszewski, M., Frellsen, J., Johansson, K. E., Hamelryck, T. (2010). 
                    <b>Beyond rotamers: a generative, probabilistic model of side chains in proteins</b>.
                    BMC Bioinformatics, 11(1), 1–13.  

                    <li> Hamelryck, T., Mardia, KV., Ferkinghoff-Borg, J., Editors. (2012) 
                    <b>Bayesian methods in structural bioinformatics.</b> 
                    Book in the Springer series "Statistics for biology and health", 
                    385 pages, 13 chapters. Springer Verlag, March, 2012

                    <li> Golden, M., Garcı́a-Portugués, E., Sørensen, M., Mardia, K. V., Hamelryck, T., Hein, J. (2017). 
                    <b>A generative angular model of protein structure evolution</b>.
                    Molecular Biology and Evolution, 34(8), 2085–2100. 
                </ul>

                <p class="aligncenter">
                    <img src="torusdbn.png" alt="torusdbn" style="width: 90vw; min-width: 100px; max-width: 700px">
                </p>

                <p>

                Sippl-type "potentials of mean force" are widely used in simulation of protein folding (AlphaFold1
                made use of them, for example). These
                potentials are typically justified by analogy with the (well-defined) potentials of mean force
                used in the physics of liquids. This is incorrect. We showed that these potentials are actually 
                an application of <b>Jeffrey's conditioning</b>, a special form of Bayesian updating. This led
                to the formulation of the <b>reference ratio method</b>, which can be used to update probabilistic models
                of local protein structure using nonlocal models (for example, on pairwise distances). This is a fine
                piece of probabilistic reasoning with many potential applications. We applied this method to 
                probabilistic modelling of structure ensembles of felxible proteins.     
                <p>
                <ul>
                    <li> Hamelryck, T., Borg, M., Paluszewski, M., Paulsen, J., Frellsen, J., Andreetta, C., Boomsma, W., Bottaro, S., Ferkinghoff-Borg, J. (2010). 
                    <b>Potentials of mean force for protein structure prediction vindicated, formalized and generalized.</b> 
                    PloS One, 5(11), e13714.

                    <li> Olsson, S., Frellsen, J., Boomsma, W., Mardia, KV., Hamelryck, T. (2013) 
                    <b>Inference of structure ensembles of flexible biomolecules from sparse, averaged data.</b> 
                    PLoS ONE. 8(11): e79439. 

                    <li> Valentin, J. B., Andreetta, C., Boomsma, W., Bottaro, S., Ferkinghoff-Borg, J., Frellsen, J., Mardia, K. V., Tian, P., Hamelryck, T. (2014). 
                    <b>Formulation of probabilistic models of protein structure in atomic detail using the reference ratio method.</b>
                     Proteins: Structure, Function, and Bioinformatics, 82(2), 288–299.
                </ul>
                <p>
                <p class="aligncenter">
                    <img src="rrm.png" alt="reference ratio" style="width: 90vw; min-width: 100px; max-width: 700px">
                </p>
                <p>
                I also edited a book on the topic of Bayesian methods in structural biology for Springer's "Statistics for biology and health" series.
                <p>

                <p class="aligncenter">
                    <img src="bmsb.png" alt="book" style="width: 33vw; min-width: 100px; max-width: 200px">
                </p>


                <h3> Deep probabilistic programming</h3>
                Deep probabilistic programming combines the modelling scope of deep learning with the principled treatment of uncertainty 
                of Bayesian statistics. My group has applied this new paradigm in machine learning to <b>Bayesian protein structure superposition</b>,
                a deep generative model of local protein structure (used in <b>vaccine design</b>), and the <b>reconstruction of ancestral 
                protein sequences</b>. 
                The latter makes use of a deep model of protein evolution based on an <b>Ornstein-Uhlenbeck process on a phylogenetic tree</b> (see
                figure below). The model can be interpreted as an ensemble of variational autoencoders whose latent variables diffuse on 
                a phylogenetic tree.
                Deep probabilistic programming and its applications is the current focus of my research group.
                <p>
                <ul>
                <li>Moreta, LS., Al-Sibahi, AS., Theobald, D., Bullock, W., Rommes, BN., Manoukian, A., Hamelryck, T. (2019) 
                    <b>A probabilistic programming approach to protein structure superposition.</b>
                     <a href="https://ieeexplore.ieee.org/document/8791469">
                     16th IEEE International Conference on Computational Intelligence in Bioinformatics and Computational Biology (CIBCB), July 9-11, 2019 – Certosa di Pontignano, Siena - Tuscany, Italy.</a>

                <li>Thygesen, CB., Al-Sibahi, AS., Steenmanns, CS., Moreta, LS., Sørensen, AB., Hamelryck, T. (2021)  
                    <b>Efficient generative modelling of protein structure fragments using a Deep Markov Model.</b> 
                    <a href="http://proceedings.mlr.press/v139/thygesen21a.html">
                    38th International Conference on Machine Learning (ICML), 139, 10258-10267</a>

                <li>Moreta, LS., Rønning, O., Al-Sibahi, AS., Hein, J., Theobald, D., Hamelryck, T. 
                    <b>Ancestral protein sequence reconstruction using a tree-structured Ornstein-Uhlenbeck variational autoencoder.</b> (2022) 
                    <a href="https://openreview.net/pdf?id=FZoZ7a31GCW">
                    ICLR 2022: The Tenth International Conference on Learning Representations.</a>
                </ul>
                <p>
                <p class="aligncenter">
                    <img src="draupnir_model.png" alt="deep evolutionary model" style="width: 90vw; min-width: 100px; max-width: 700px">
                </p>

                </div> 
            </div> 

            <!-- Contact Info on the Sidebar -->
            <div class="col-md-4">
                <div style="font-family: 'Oswald', sans-serif; font-size: 32px;"><b>Thomas Hamelryck</b></div><br>
                <p><b>thamelry@bio.ku.dk</b><br>
                <p>Department of biology<br>
                Section for Computational and RNA Biology (SCARB) <br>
                Ole Maaløes Vej 5 <br>
                DK-2200 Copenhagen N <br>
                <p>and <br>
                <p>Department of computer science<br>
                Programming languages and theory of computation section (PLTC) </br>
                Universitetsparken 5, HCØ, building B <br>
                DK-2100 Copenhagen Ø <br>
                <p>University of Copenhagen<br>
                Denmark<br>
                </p>
            </div>
            
            
            <!-- Links on the Sidebar -->
            <div class="col-md-4" style="margin-top:2%">
              <dd><a href="https://scholar.google.dk/citations?hl=en&user=YoTlzjkAAAAJ">Google Scholar</a></dd> 
              <dd><a href="https://www.linkedin.com/in/thomas-hamelryck-41a0a64/">LinkedIn</a></dd>
            </div>


        </div>


    </div>
    <!-- /.container -->
    
    <!-- Other people may like it too! -->
    <a style="color:#b5bec9;font-size:0.8em; float:right;" href="https://github.com/mavroudisv/plain-academic">Plain Academic</a> 
    
</body>

</html>
